lib/truffle/truffle/patching.rb

```diff
@@ -78,4 +78,21 @@ def activate
     end
   end
 
+  # Allows specifying patching outside the context of RubyGems by setting an environment variable value.
+  # E.g., setting TRUFFLERUBY_CUSTOM_PATCH='launchy:lib,spec' will allow patching of files in the 'lib' and 'spec'
+  # of a local checkout of the 'launchy' gem.
+  def install_local_patches
+    custom_patch = Truffle::POSIX::NATIVE && ENV['TRUFFLERUBY_CUSTOM_PATCH'] # We can't read from ENV if native is disabled.
```

Comment:
You could read with System.getenv() in Java and expose it as a primitive, I think that would be fine and less costly than reading with `ENV[]` on every startup.
Or it could be an option, which would be a bit cleaner but maybe harder to pass around if TRUFFLERUBYOPT is overridden.

Reply:
An extra `ENV[]` slows down startup by ~1-2ms iirc, and we might be able to use it in other places.